Core/Src/main.c

```diff
@@ -87,16 +89,56 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // The data to transmit.
+  // The 0b1 indicates a start bit.
+  // The 0b10000000 selects CH0 of the ADC, single-ended.
+  // The 0b0 is really a don't care for the "sample and hold" period.
+  uint8_t txBuffer[3] = {0b1, 0b10000000, 0b0};
+
+  // The buffer to receive the data.
+  uint8_t rxBuffer[3] = {0};
+
+  // The largest value the ADC can output (2^10 - 1).
+  const uint16_t maxAdcVal = 1023;
+
+  // Min and max PWM percentages for duty cycle.
+  const float minPWM = 0.05;
+  const float maxPWM = 0.10;
+
+  // Start the timer for PWM.
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  // Start communication by bringing CS low.
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  // Use the ADC.
+	  HAL_SPI_TransmitReceive(&hspi1, txBuffer, rxBuffer, 3, 1000);
```

Comment:
definitely a nit pick, but 3 and 1000 would be considered magic numbers… if I wasn’t familiar with the HAL_Spi_TransmitReceive function then i would have no idea what those values mean. May be best to create a #define macro for them.

Reply:
Done.